crates/biome_js_analyze/src/analyzers/nursery/no_useless_ternary.rs

```diff
@@ -88,6 +112,89 @@ impl Rule for NoUselessTernary {
+            .value_token()
+            .ok()?;
+        let consequent = node
+            .consequent()
+            .ok()?
+            .as_any_js_literal_expression()?
+            .as_js_boolean_literal_expression()?
+            .value_token()
+            .ok()?;
+
+        let new_node;
+        let mut mutation = ctx.root().begin();
+
+        if alternate.text_trimmed() == consequent.text_trimmed() {
+            if node_expression_kind == JsSyntaxKind::JS_IDENTIFIER_EXPRESSION {
+                let res_boolean_value = if consequent.text_trimmed() == "true" {
+                    T![true]
+                } else {
+                    T![false]
+                };
+
+                new_node = biome_js_syntax::AnyJsExpression::AnyJsLiteralExpression(
+                    AnyJsLiteralExpression::JsBooleanLiteralExpression(
+                        make::js_boolean_literal_expression(make::token(res_boolean_value)),
+                    ),
+                );
+            } else {
+                return None;
+            }
+        } else if alternate.text_trimmed() == "true" {
+            new_node = invert_expression(&node.test().ok()?)?;
+        } else if is_boolean_expression(&node.test().ok()?)? {
+            match node_expression_kind {
+                JsSyntaxKind::JS_BINARY_EXPRESSION => {
+                    let left = node.test().ok()?.as_js_binary_expression()?.left().ok()?;
+                    let right = node.test().ok()?.as_js_binary_expression()?.right().ok()?;
+                    let operator = node
+                        .test()
+                        .ok()?
+                        .as_js_binary_expression()?
+                        .operator_token()
+                        .ok()?
+                        .kind();
+
+                    new_node = AnyJsExpression::from(make::js_binary_expression(
+                        left,
+                        make::token_decorated_with_space(operator),
+                        right,
+                    ));
+                }
+                JsSyntaxKind::JS_UNARY_EXPRESSION => {
+                    let argument = node
+                        .test()
+                        .ok()?
+                        .as_js_unary_expression()?
+                        .argument()
+                        .ok()?;
+                    new_node = make::js_unary_expression(make::token(T![!]), argument).into();
+                }
+                _ => return None,
+            }
+        } else {
+            let new_expression = invert_expression(&node.test().ok()?)?;
+            new_node = make::js_unary_expression(make::token(T![!]), new_expression).into();
+        }
+
+        mutation.replace_element(node.clone().into(), new_node.into());
+        return Some(JsRuleAction {
+            category: ActionCategory::QuickFix,
+            applicability: Applicability::MaybeIncorrect,
+            message: markup! { "Replace the above code with," }.to_owned(),
```

Comment:
```suggestion
            message: markup! { "Remove the binary expression" }.to_owned(),
```

Reply:
But it's not always a binary expression.
I think it's better -> Replace the conditional expression with

Reply:
yeah that works! Just saying "replace code" is too generic and redundant because it's a code action :)